docs/fsharp-cheatsheet.md

```diff
@@ -740,6 +743,112 @@ Another way of implementing interfaces is to use *object expressions*.
 
 *Partial active patterns* share the syntax of parameterized patterns but their active recognizers accept only one argument.
 
+<div id="asynchronous-programming"></div>
+
+## Asynchronous Programming
+
+F# asynchronous programming is centered around two core concepts: async computations and tasks.
+
+    async {
+        do! Async.Sleep (waitInSeconds * 1000)
+        let! asyncResult = asyncComputation
+        use! disposableResult = iDisposableAsyncComputation
+    }
+
+### Async vs Tasks
+
+An async computation is a unit of work, and a task is a promise of a result. A subtle but important distinction.
+Async computations are composable and are not started until explicitly requested; Tasks (when created using the `task` expression) are _hot_:
+
+    let runAsync waitInSeconds =
+        async {
+            printfn "Created Async"
+            do! Async.Sleep (waitInSeconds * 1000)
+            printfn $"Completed Async"
+        }
+        
+    let runTask waitInSeconds =
+        task {
+            printfn "Started Task"
+            do! System.Threading.Tasks.Task.Delay (waitInSeconds * 1000)
+            printfn $"Completed Task"
+        }
+    
+    let asyncComputation = runAsync 5 // returns Async<unit> and does not print anything
+    let newTask = runTask 3  // returns System.Threading.Tasks.Task<unit> and outputs: "Started Task" <3 second delay> "Completed Task"
+
+    asyncComputation |> Async.RunSynchronously  // this now runs the async computation
+    newTask.Wait() // this will have already completed by this point
+
+    // Output:
+    // Started Task
+    // Created Async
+    // Completed Task
+    // Completed Async
+
+### Async and Task Interop
+
+As F# sits in .NET, and a lot of the codebase uses the C# async/await, the majority of actions are going to be executed and tracked using `System.Threading.Tasks.Task<'T>`s.
+
+#### Async.AwaitTask
+
+This converts a Task into an async computation. It has the [signature](#functions-signatures): `Task<'T>` -> `Async<'T>`
+
+    async {
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn $"File {fileName} has %d{bytes.Length} bytes"
+    }
+
+#### Async.StartAsTask
+
+This converts an async computation into a Task. It has the [signature](#functions-signatures): `Async<'T>` -> `Task<'T>`
```

Comment:
While this is not untrue, it's also not very useful information - there's lots of discussion as to why in this PR, and we landed on this wording:
https://github.com/fsprojects/FSharp.Control.TaskSeq/pull/249/files#diff-0d723d41a6a5de091170b77b03f31ca3954f6a038be6d72fe42f00bd6f19d6f2R96

Reply:
<ramble>
Here again it depends how much pre-explanation you've done what you say here....
The other elephant is that StartAsTask does the starting in the threadpool, whereas normally IRL you want to do Async.StartImmediateAsTask to kick off some work in the context of a computation expression.
Thats kinda entirely irrelevant and overly detailed, but if I was going to only mention one I think I'd go for StartImmediateAsTask.
I would always show that normally you want to pass in a cancellatino token
And you might show that cancelling it would cut short the wait on L808
</ramble>
----
As this is, I would say something like that this kicks of an instance of the workflow (without an attached cancellation token). But I don't think I'd use this example in the first instance though.
And I'd be seeking to convey that you can do it N times

I'd maybe also make an async with no arguments and show that you can run it N times - building (pardon the pun) up the concepts more slowly.

Reply:
Somewhere you also need to cover that e.g. a Task that has faulted will cache the exception result, whereas each Async.RunSynchronously has an independent execution - i.e. the first could throw and the second succeed etc